helper/memory_calculator.go

```diff
@@ -103,7 +106,22 @@ func (m MemoryCalculator) Execute() (map[string]string, error) {
 	}
 
 	if totalMemory == UnsetTotalMemory {
-		m.Logger.Info("WARNING: Container memory limit unset. Configuring JVM for 1G container.")
+		if b, err := ioutil.ReadFile(m.MemoryInfoPath); err != nil && !os.IsNotExist(err) {
+			m.Logger.Info("Unable to read %s\n%w", m.MemoryInfoPath, err)
+		} else if err == nil {
+			rp := regexp.MustCompile("MemAvailable:\\s*(\\d*)\\s")
+			s := rp.FindStringSubmatch(string(b))[1]
+
+			if i, err := strconv.ParseInt(s, 10, 64); err != nil {
+				return nil, fmt.Errorf("untable to convert available memory %s to integer\n%w", s, err)
+			} else {
+				totalMemory = i * calc.Kibi
```

Comment:
Although in practice it appears that `MemAvailable` is usually in units of `kB`, based on the `meminfo` documentation, I don't think we can assume that it is always `kB`. I think we should either check the units or at least at `kB` to our regex.